### PR TITLE
Update README and Documentation

### PR DESCRIPTION
**Pull request by arjunsalyan**
- [x] README
- [x] Deployment
- [x] API
- [x] Collection of Data
- [x] History
- [x] About page

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,152 +7,31 @@ This is a web application that:
  2. Fetches and displays build history of the ports.
  3. Collects and displays installation statistics (system configurations and installed ports) from users who opt-in by
     installing the port `mpstats`.
+ 4. Allows to follow ports and receive notifications when a port gets updated.
```
Comment (mojca) [inline README.md]: Maybe
```suggestion
 4. Allows following ports and receiving notifications when a port gets updated.
```
?

File: docs/DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+Download and install `Postgres.app` from [here](https://postgresapp.com/downloads.html).
+
+After this, connect as root:
+```bash
+su - postgres
+```
+
+Create your user(replace username and password with your preferred credentials):
+```bash
+CREATE USER username PASSWORD 'passowrd';
```
Comment (mojca) [inline docs/DEPLOYMENT.md]: Not that it really matters, but:
```suggestion
CREATE USER username PASSWORD 'password';
```

File: docs/DEPLOYMENT.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+
+The external dependencies of the app are: PostgreSQL, Solr and Memcached. The search page(/search/) is completely derived from
+Solr. Memcached is optional, and can be avoided if you don't want to cache the results of stats and trac tickets.
+
+If you are new to these technologies, you may use the instructions provided below, otherwise you are free to install and setup these
+dependencies using your preferred mechanism.
+
+### PostgreSQL
+
+Download and install `Postgres.app` from [here](https://postgresapp.com/downloads.html).
```
Comment (mojca) [inline docs/DEPLOYMENT.md]: We could also say
> Either download and install `Postgres.app` from [here](https://postgresapp.com/downloads.html) or via MacPorts:
> `sudo port install postgresql13 postgresql13-server`

File: docs/DEPLOYMENT.md [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+Create a new database;
+
+```bash
+CREATE DATABASE webapp;
+```
+
+Grant permissions to your user on this database.
+
+```bash 
+GRANT ALL PRIVILEGES ON DATABASE webapp TO username;
```
Comment (mojca) [inline docs/DEPLOYMENT.md:55]: This is ok for playing around, but it's not suitable for the final deployment. I'm not sure whether we need to be picky on this.

File: docs/DEPLOYMENT.md [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+That's it for PostgreSQL, our database for the app is ready.
+
+### Solr
+
+To get the required version of Solr working we need [openjdk8](https://openjdk.java.net/install/). On a mac, you may use 
+[MacPorts to install openjdk8](https://ports.macports.org/port/openjdk8/). New versions of openjdk may raise problems,
+we recommend openjdk8.
+
+Once opendjk8 has been installed, we can proceed to downloading and launching Solr. We would need 
+[Solr 6.6.6](https://archive.apache.org/dist/lucene/solr/6.6.6/). The below steps are self explanatory:
```
Comment (mojca) [inline docs/DEPLOYMENT.md:67]: I see that we provide `apache-solr8` as a package as well. If we keep mentioning macports, we could probably mention this port as well, unless it's not suitable for some reason.

Just a question. I see that version 8.9.0 is out. This one points to 6.6.6. Is there some reason to use outdated version (other than the fact that these docs were written long time ago)?

File: docs/DEPLOYMENT.md [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+    --name=webapp-container \
+    --env-file=env \ 
+    macports-webapp
+```
+
+Open `127.0.0.1:8080` in the browser to test if the app has started, you won't have any ports though.
+
+### 3. Setting up initial data
+These instructions are common for the docker and without docker, so please skip the next section and jump to
+"Initialising the web app"
```
Comment (mojca) [inline docs/DEPLOYMENT.md]: This section sounds relatively confusing to me.

You could add a sentence (without its own section) like:
```
Continue with [Initialising the Webapp](#initialising-the-webapp).
```

File: docs/DEPLOYMENT.md [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+
+You may now run the server, but your app has no data, yet.
+
+```bash
+python3 manage.py runserver
+```
+
+## Initialising the Webapp
+
+Now that you have the app running(using either of two routes above), it is time to add some data to the app. Some of these
```
Comment (mojca) [inline docs/DEPLOYMENT.md]: This applies to a couple of more places. Opening parenthesis should almost always be preceded by whitespace.  
```suggestion
Now that you have the app running (using either of two routes above), it is time to add some data to the app. Some of these
```

File: README.md [hunk 8]
```diff
@@ -7,152 +7,31 @@ This is a web application that:
...
+The web application is based on:
+ - [Django Framework](https://www.djangoproject.com)
+ - [PostgreSQL Database](https://www.postgresql.org)
+ - [Solr](https://lucene.apache.org/solr/)
+ - [Memcached](http://memcached.org)
  
-This web application has been implemented during the [Google Summer of Code 2019](https://summerofcode.withgoogle.com) 
-mentored by the [MacPorts](https://www.macports.org) organisation.
-
-**Student**: [Arjun Salyan](https://github.com/arjunsalyan) <br>
```
Comment (mojca) [inline README.md:25]: So you removed your name and kept `HISTORY.md` empty?

Comment (arjunsalyan) [reply README.md:25]: I was simplifying the readme with all the information in separate links. But never really got to the point of finishing all the files. I hope to be able to do it soon.

File: docs/DEPLOYMENT.md (hunk 5 above)
Comment (arjunsalyan) [reply docs/DEPLOYMENT.md:67]: `6.6.6` is the latest django-haystack supports. I had this discussion with Amar last year during the development phase and he said that is fine.

Comment (mojca) [reply docs/DEPLOYMENT.md:67]: Sure, that's fine. It's just probably helpful to explicitly mention the reason for sticking with a particular version when that's the case. Else one might decide to update everything one day and waste time debugging why things don't work.

File: README.md (hunk 8 above)
Comment (mojca) [reply README.md:25]: I'm aware that the docs were not finished, and I don't expect them to be complete. It's just strange that you completely remove your name ;)

**Pull request merged**